backend/routes/api/users.js

```diff
@@ -0,0 +1,166 @@
+const express = require("express");
+const router = express.Router();
+const User = require("../../models/User");
+const { celebrate, Joi } = require("celebrate");
+// get all users
+router.get(
+  "/",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string().required(),
+      lastName: Joi.string().required(),
+      oauthId: Joi.string().required(),
+      propicUrl: Joi.string().required(),
+      isApproved: Joi.boolean().default(false),
+      role: Joi.string().required(),
+      location: Joi.string().required()
+    })
+  }),
+  async (req, res) => {
+    const users = await User.find({});
+    res.json({
+      code: 200,
+      result: users,
+      success: true
+    });
+  }
+);
+
+// get one user
+router.get("/:user_id", async (req, res) => {
+  const userId = req.params.user_id;
+  const user = await User.findById(userId);
+  res.json({
+    code: 200,
+    result: user,
+    success: true
+  });
+});
+
+// create new user
+router.post(
+  "/",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string().required(),
+      lastName: Joi.string().required(),
+      oauthId: Joi.string().required(),
+      propicUrl: Joi.string().required(),
+      isApproved: Joi.boolean().default(false),
+      role: Joi.string().required(),
+      location: Joi.string().required()
+    })
+  }),
+  async (req, res) => {
+    const data = req.body;
+    const newUser = new User({
+      firstName: data.firstName,
```

Comment:
We *probably* won't be using this endpoint in the near future given that we'll be doing all the creation through OAuth as of now, but I'm glad that you have it. Definitely part of the CRUD specs to have this in here.